frontend/src/metabase/search/analytics.ts

```diff
@@ -0,0 +1,35 @@
+import { trackSchemaEvent } from "metabase/lib/analytics";
+import type { URLSearchFilterQueryParams } from "metabase/search/types";
+
+export const trackSearchEvents = ({
+  searchFilters,
+}: {
+  searchFilters: URLSearchFilterQueryParams;
+}) => {
+  let content_type = null;
+
+  if (searchFilters.type) {
+    content_type = Array.isArray(searchFilters.type)
+      ? searchFilters.type
+      : [searchFilters.type];
+  }
+
+  const filterBooleanPayload: { [key: string]: unknown } = {
+    creator: searchFilters.created_by,
+    last_editor: searchFilters.last_edited_by,
+    creation_date: searchFilters.created_at,
+    last_edit_date: searchFilters.last_edited_at,
+    verified_items: searchFilters.verified,
+    search_native_queries: searchFilters.search_native_query,
+  };
+
+  Object.entries(searchFilters).forEach(([key, value]) => {
+    filterBooleanPayload[key] = !!value;
+  });
```

Comment:
Instead of casting them to boolean here, what about casting them since the creation of `filterBooleanPayload`? It's not like we're casting them based on any condition or anything here.

```
  const filterBooleanPayload: { [key: string]: unknown } = {
    creator: !!searchFilters.created_by,
    last_editor: !!searchFilters.last_edited_by,
    creation_date: !!searchFilters.created_at,
    last_edit_date: !!searchFilters.last_edited_at,
    verified_items: !!searchFilters.verified,
    search_native_queries: !!searchFilters.search_native_query,
  };
```

Although I still find using the boolean constructor more readable e.g. `Boolean(searchFilters.created_by)`, `!!` is still fine for me.

Reply:
I think the constructor is the best one here. I actually had it like your suggestion before, but I didn't want to repeat myself too much, but I guess with this case where it's set in stone it's ok